### ISSUE TITLE
Add Aesara as a backend

### ISSUE THREAD

**Issue by rlouf** [dgasmith/opt_einsum#213]
I am a core member of [Aesara](https://github.com/aesara-devs/aesara), a fork of the now deprecated Theano. Since there is a Theano backend, do you think it would make sense to have an Aesara backend as well?

**Comment by dgasmith** [comment]
@rlouf Absolutely, always happy to review PRs!

**Comment by jcmgray** [comment]
Yes it would certiainly be nice to have, I'll just note the requirements are pretty simple:

- [x] ability to dynamically define high dimensional tensors.
- [x] `tensordot` implementation
- [ ] (optional) `einsum` implementation -- without this only some contractions can be performed

**Comment by jcmgray** [comment]
Possibly relevant question @rlouf, what is the best way to get the shape of a `TensorType` as a plain tuple of int? They don't have an `infer_shape` method. Is `t.shape.eval()` the way to go?

**Comment by rlouf** [comment]
You can try `t.type.shape` if the shape is known before evaluating the graph.
